x-pack/plugins/ml/public/application/timeseriesexplorer/components/plot_function_controls/plot_function_controls.tsx

```diff
@@ -30,11 +35,70 @@ const plotByFunctionOptions = [
 export const PlotByFunctionControls = ({
   functionDescription,
   setFunctionDescription,
+  selectedDetectorIndex,
+  selectedJobId,
+  selectedEntities,
 }: {
   functionDescription: undefined | string;
   setFunctionDescription: (func: string) => void;
+  selectedDetectorIndex: number;
+  selectedJobId: string;
+  selectedEntities: Record<string, any>;
 }) => {
+  const toastNotificationService = useToastNotificationService();
+
+  const getFunctionDescriptionToPlot = useCallback(
+    async (
+      _selectedDetectorIndex: number,
+      _selectedEntities: Record<string, any>,
+      _selectedJobId: string,
+      _selectedJob: CombinedJob
```

Comment:
I believe you don't need it and can access these arguments from the component scope

Reply:
I think doing that will require the callback to depend on those arguments, and since they are the same as the original effect we're splitting it from, I don't see the benefits of either moving it outside of the effect or using the arguments from component scope.